qrexec/policy/parser.py

```diff
@@ -550,7 +550,7 @@ class AbstractResolution(metaclass=abc.ABCMeta):
     """Object representing positive policy evaluation result -
     either ask or allow action"""
 
-    def __init__(self, rule, request, *, user):
+    def __init__(self, rule, request: Request, *, user: str):
```

Comment:
`Request` isn't declared at this point. But you can use [forward reference](https://peps.python.org/pep-0484/#forward-references) (`'Request'`), or `from __future__ import annotations`